pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -837,14 +837,14 @@ func applyLocalityLBSetting(
 		if shared {
 			// update the locality settings only if the cluster has
 			// outlier detection settings
-			if cluster.LoadAssignment != nil && cluster.OutlierDetection != nil {
```

Comment:
Actually, with `OutlierDetectio ` it is better or envoy to adjust the weight per locality.
But i agree this is not a must.

Reply:
Not sure what you mean by this?

Reply:
I mean it does not matter